### PR TITLE
Add an executor that can deploy to Docker swarm

### PR DESCRIPTION
**Pull request by stjohnjohnson**
Supports local docker as well (blocked on log-service rewrite)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+                    HostConfig: {
+                        Memory: 2 * 1024 * 1024 * 1024,
+                        MemoryLimit: 3 * 1024 * 1024 * 1024,
+                        VolumesFrom: [
+                            `${launchContainer.id}:rw`
+                        ]
+                    }
+                })
+            ]))
+            .then(([logContainer, buildContainer]) => Promise.all([
```
Comment (d2lam) [inline index.js]: Ahh this is the thing you were talking about. It would have to wait to create and start both containers.

File: index.js [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+     * Stop a s3m build
+     * @method _stop
+     * @param  {Object}   config            A configuration object
+     * @param  {String}   config.buildId    ID for the build
+     * @return {Promise}
+     */
+    _stop(config) {
+        return this._findContainers(config.buildId)
+            .then(containers => Promise.all(containers.map(container =>
+                this._removeContainer(container))));
```
Comment (tkyi) [inline index.js]: Would there be any harm in removing containers before stopping them?

Comment (stjohnjohnson) [inline index.js]: Since we're passing force to it (it's in the function), it will stop and then remove it.

File: index.js [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+     * @param  {String} [options.docker.protocol]                Protocol to use
+     * @param  {String} [options.docker.host]                    Docker Swarm host to interact with
+     * @param  {String} [options.docker.port]                    Port number
+     * @param  {String} [options.docker.socketPath]              Docker socket to use
+     * @param  {String} [options.docker.ca]                      Certificate authority
+     * @param  {String} [options.docker.cert]                    Certificate to use
+     * @param  {String} [options.docker.key]                     Key for the certificate
+     * @param  {Object} [options.fusebox]                        Fusebox configuration
+     * @param  {Object} [options.fusebox.breaker]                Breaker configuration
+     * @param  {Object} [options.fusebox.breaker.timeout=300000] Timeout before retrying
```
Comment (FenrirUnbound) [inline index.js]: Based on the default value, this should be number `Number`

File: index.js [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+     * @return {Promise}
+     */
+    _stop(config) {
+        return this._findContainers(config.buildId)
+            .then(containers => Promise.all(containers.map(container =>
+                this._removeContainer(container))));
+    }
+
+    /**
+    * Retreive stats for the executor
```
Comment (FenrirUnbound) [inline index.js]: For the executor itself? Or are you referring to the Fusebox?

File: index.js [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+                        '--emitter',
+                        '/opt/screwdriver/emitter',
+                        config.buildId
+                    ],
+                    Env: [
+                        `SD_TOKEN=${config.token}`
+                    ],
+                    HostConfig: {
+                        Memory: 200 * 1024 * 1024,
+                        MemoryLimit: 300 * 1024 * 1024,
```
Comment (FenrirUnbound) [inline index.js]: Might be worth mentioning what units these are in.

As an aside, what's the difference between the two settings?

**Pull request merged**